metadata-structure/metadata-structure.md

```diff
@@ -0,0 +1,340 @@
+1. `value` is an optional key to set the default value of the `control` as something other than the default (default in frontend for single select is the first value and for a multi-select is all the values). This is the `id` of the option (`a` in the visualisation)
+
+## Settings
+
+```php
+├── filterTypes[]
+│   ├── id
+│   ├── column_id
+│   ├── use_shape_regions
+│   ├── options[]
+│   │   ├── label
+│   │   ├── id
+│   │   ├── description
+├── indicators
+├── plotSettingsControl{}
+│   ├── defaultEffect
+│   │   ├── setFilters
+│   │   ├── ... (see effect below)
+│   ├── plotSettings[]
+│   │   ├── label
+│   │   ├── id
+│   │   ├── visible
+│   │   ├── value
+│   │   ├── options[] # here
+│   │   │   │   ├── id # 1
+│   │   │   │   ├── label # 2
+│   │   │   │   ├── effect # 3
+│   │   │   │   │   ├── setFilters
+│   │   │   │   │   ├── setMultiple
+│   │   │   │   │   ├── setFilterValues
+│   │   │   │   │   ├── setHidden
+```
+
+The `settings` completely determine the `filters` and their `selections`. The next section will go through each effect and how the frontend resolves it but for now, explanation of keys:
+
+1. `id` for the store to record what `setting` is chosen
+1. `label` displayed to the user in the dropdown menu
+1. The `effect` key is probably the most important idea here. Every `setting` has an effect associated with it and aggregating all the effects along all the settings the user has chosen completely determines the `filters` and their initial `selections` that are shown to the user. So everything can be derived from just the `settings` that are chosen
+
+## Effects
+
+```php
+├── filterTypes[]
+│   ├── id # a
+│   ├── column_id
+│   ├── use_shape_regions
+│   ├── options[]
+│   │   ├── label
+│   │   ├── id # b
+│   │   ├── description
+├── indicators
+├── plotSettingsControl{}
+│   ├── defaultEffect # (here as well)
+│   │   ├── setFilters
+│   │   ├── ... (see effect below)
+│   ├── plotSettings[]
+│   │   ├── label
+│   │   ├── id
+│   │   ├── visible
+│   │   ├── value
+│   │   ├── options[]
+│   │   │   │   ├── id
+│   │   │   │   ├── label
+│   │   │   │   ├── effect # here
+│   │   │   │   │   ├── setFilters
+│   │   │   │   │   ├── setMultiple
+│   │   │   │   │   ├── setFilterValues
+│   │   │   │   │   ├── setHidden
+```
+
+For this section we will need a more detailed view of the effects structure
+
+```php
+├── setFilters[] # 1
+│   ├── filterId # 1.1
+│   ├── label # 1.2
+│   ├── stateFilterId # 1.3
+├── setMultiple[] # 2
+├── setFilterValues{} # 3
+├── setHidden[] # 4
+```
+
+Explanation of each effect and how the frontend aggregates this:
+1. `setFilters` is how the effect specifies what `filters` the user sees from the `filterTypes` The frontend aggregates this by replacing it. So if two `settings` set filters, the last `setFilters` in the for loop would be used when going through all the `settings`. Keys:
```

Comment:
```suggestion
1. `setFilters` is how the effect specifies what `filters` the user sees from the `filterTypes`. The frontend aggregates this by replacing it. So if two `settings` set filters, the last `setFilters` in the for loop would be used when going through all the `settings`. Keys:
```

Adds full stop

Reply:
Updated this, the schema already limits this to just 1 per control but can't do that across multiple controls. Updating wording a little